src/py/flwr/common/typing.py

```diff
@@ -208,23 +208,44 @@ class ClientMessage:
 
 
 @dataclass
-class Run:
+class RunStatus:
+    """Run status information."""
+
+    status: str
+    sub_status: str
+    details: str
+
+
+@dataclass
+class Run:  # pylint: disable=too-many-instance-attributes
     """Run details."""
 
     run_id: int
     fab_id: str
     fab_version: str
     fab_hash: str
     override_config: UserConfig
-
-
-@dataclass
-class RunStatus:
-    """Run status information."""
-
-    status: str
-    sub_status: str
-    details: str
+    pending_at: str
+    starting_at: str
+    running_at: str
+    finished_at: str
+    status: RunStatus
+
+    @classmethod
+    def create_empty(cls, run_id: int = 0) -> "Run":
```

Comment:
Should `run_id` have a default? Or should we force the caller to set it?

Reply:
Hmmm, I think that's fine. The purpose of this class method is just to create an empty instance, and it's not a constructor.